users/users.go

```diff
@@ -18,6 +23,19 @@ type User struct {
 // DefaultDataProvider implements the DataProvider interface.
 type DefaultDataProvider struct{}
 
+// FindUser finds user by ID, which could be either an integer or a user name.
+func (t *DefaultDataProvider) FindUser(id string) (*User, error) {
```

Comment:
It seems that Go [does not support method overloading](https://golang.org/doc/faq#overloading) but adding this logic inside this method would restrict userIDs to not be numeric. (I don't know if that really matters, but could be something to consider).

Reply:
There are two other methods for that, `FindUserByUserName()` and `FindUserByID()`. This method handles both cases just for convenience.

Reply:
Yeah, for me it's not a big deal as long as it is not the only way to retrieve users.